src/authorization.rs

```diff
@@ -1,19 +1,20 @@
-use oauth2::PkceCodeChallenge;
-use openidconnect::{CsrfToken, IssuerUrl};
+use std::{borrow::Cow, marker::PhantomData};
+
+use oauth2::{CsrfToken, PkceCodeChallenge};
 use serde::{Deserialize, Serialize};
 use url::Url;
 
-use crate::profiles::AuthorizationDetailsProfile;
+use crate::{
+    profiles::AuthorizationDetailsProfile,
+    types::{IssuerState, IssuerUrl, UserHint},
+};
 
 pub struct AuthorizationRequest<'a, AD>
 where
     AD: AuthorizationDetailsProfile,
 {
-    inner: oauth2::AuthorizationRequest<'a>, // TODO
-    authorization_details: Vec<AuthorizationDetail<AD>>,
-    wallet_issuer: Option<IssuerUrl>, // TODO SIOP related
-    user_hint: Option<String>,
-    issuer_state: Option<CsrfToken>,
+    inner: oauth2::AuthorizationRequest<'a>,
+    _pd: PhantomData<AD>,
```

Comment:
I don't think it's necessary to keep `AD` in this type? It can simply be a generic of `set_authorization_details`